Automattic/wp-calypso
# ImportNux: Add validation to url input field

> Note that this PR points to #26502, the master feature branch for the importer NUX

This PR adds some basic validation to our URL input field found in the Importer Nux flow.
Currently, we check only that the URL is either valid or invalid, showing a generic, non-nuanced validation message when an invalid value is typed.

The 'Continue' button should now also reflect the input - when it's empty or has a validation error, it's disabled.

I've made some effort here to give an improved experience to those using this input, in terms of how and when the validation is shown. The basic idea here is that I want to give customers the chance to finish typing before showing any validation. As such I've debounced setting the validation message where there is one. However,  when there is no message to show, when the input is valid, we set this immediately to give instant feedback that everything is 'OK'.
Let me know if this logic makes sense or could be improved. It's something I'd love to nail down as a solid pattern to allow us to make improvements elsewhere in Calypso.

One last note - I've repurposed the debouncing from the `fetchIsSiteImportable` logic, which was removed. This behaviour will be re-implemented soon in the next step of this signup flow.

### To Test

- ~Needs D17199-code & a patched API sandbox for now~
- Run this branch locally
- Visit http://calypso.localhost:3000/start/from-site
- It should redirect to http://calypso.localhost:3000/start/from-site/import-from-url & you should see a new step with a form to enter your URL.
- Start typing something invalid (no `.`s for example)
  - You should notice a slight delay after typing before seeing a validation message
- fix up the url to make it valid
  - As soon as you've hit the criteria, you should notice that the validation message clears and the 'continue' button becomes clickable
